lib/chef/provisioning/machine_spec.rb

```diff
@@ -39,43 +39,26 @@ def name
     #
     # chef-provisioning will do its darnedest to not lose this information.
     #
-    def location
-      chef_provisioning_attr('location')
+    def reference
+      attrs['reference'] || attrs['location']
```

Comment:
`nil` and `false` aren't valid values for it anyway, making this a bit more readable--but yes, I loves me some `fetch`.